### ISSUE TITLE
Pusher open fd

### ISSUE THREAD

**Issue by stephen-soltesz** [m-lab/pusher#72]
Pusher opens and olds file descriptiors for up to 2 hours. Unclear if this is WAI.

```
process_open_fds{machine=~"$machine.$site.*", deployment=~"$deployment"}
```

![pusher-openfds](https://user-images.githubusercontent.com/1085316/69271766-7c3a6d00-0ba3-11ea-8082-bad2bb4b2e78.png)

**Comment by stephen-soltesz** [comment]
During load testing pusher accumulated ~300k open fds. The ulimit for the container is 1M.

![pusher-300k-fd](https://user-images.githubusercontent.com/1085316/69465921-96fe1480-0d50-11ea-9706-0fa0fedb2fcf.png)

**Comment by pboothe** [comment]
This is happening because each `Tarfile` struct keeps track of its component files by saving a list of `os.File` objects. https://github.com/m-lab/pusher/blob/master/tarfile/tarfile.go#L103

We could change this to cache names rather than files - our only reason for keeping the list is to eventually call 
```go
	for _, file := range t.members {
		// If the file can't be removed, then it either was already removed or the
		// remove call failed for some unknown reason (permissions, maybe?). If the
		// file still exists after this attempted remove, then it should eventually
		// get picked up by the finder.
		if err := os.Remove(file.Name()); err == nil {
```
(https://github.com/m-lab/pusher/blob/master/tarfile/tarfile.go#L237)

and that just calls `file.Name()` to get the name.

**Comment by pboothe** [comment]
I think the correct change is to delete the `members` element entirely and change `memberSet` to be of type `memberSet  map[filename.Internal]filename.System`. Then, after a successful upload, delete every `filename.System` stored in the map.

**Comment by pboothe** [comment]
Should be fixed in #74
